File: Hash_Function.py
def hashFunc(piece):
    words = piece.split(" ") #splitting string into words
    colour = words[0]
    shape = words[1]
    poleNum = 0
    for i in range(0, 3):
        poleNum += ord(colour[i]) - 96
        poleNum += ord(shape[i]) - 96
    return poleNum

File: Hash_Function.js
/*
* Tests with:
* hash('Color red')
*/

function hash(piece) {
  var words = piece.split(" "),
    colour = words[0],
    shape = words[1],
    hashNumber = 0;
    for (var i = 0; i < 3; i++) {
      hashNumber += colour[i].charCodeAt(0) - 64;
      hashNumber += shape[i].charCodeAt(0) - 64;
    }
    return hashNumber;
}
